### ISSUE TITLE
buildpro sub branch for submodule support

### ISSUE THREAD

**Issue by smanders** [smanders/buildpro#59]
rather than copying select files from the `.devcontainer/` directory to a project that leverages buildpro, it would be nice if there was an option to add buildpro as a submodule and automatically have the files needed (no more sync'ing req'd)

the sub branch https://github.com/smanders/buildpro/tree/sub is for this purpose!

```
git submodule add -b sub https://github.com/smanders/buildpro .devcontainer
```
will add the buildpro sub branch to a `.devcontainer/` directory in a repo that wants to leverage buildpro

and if the sub branch commits are tagged with the buildpro release that is supported, the `denv.sh` script can be updated to automatically know the buildpro release

in other words, if the buildpro release is `23.04.3` and the contents of the `.devcontainer/` directory in the `dev` branch match the root directory in the `sub` branch, and a tag of `v23.04.3` exists in the `sub` branch to indicate the match, then a project using buildpro as a submodule doesn't need to specify the buildpro release

NOTE: some repos (like CommonLibraries and PluginLibraries) shouldn't use buildpro as a submodule because we want to avoid nested submodules -- so they should continue to sync the `.devcontainer/` directory of buildpro with their `.devcontainer/` directory

**Comment by smanders** [comment]
completed with commit to dev branch referenced above
